core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -6333,8 +6337,8 @@ sealed trait Exit[+E, +A] extends ZIO[Any, E, A] { self =>
     trace: Trace
   ): ZIO[R, E1, B] =
     self match {
-      case Failure(cause) => failed(cause)
       case Success(v)     => completed(v)
+      case Failure(cause) => failed(cause)
```

Comment:
Seems reasonable, but do you have any particular rationale?

Reply:
Since a Failure most of the times results in a termination, we want to optimize for the happy-path (success) so we check that one first